src/slicer.cpp

```diff
@@ -962,27 +987,50 @@ Slicer::Slicer(Mesh* mesh, const coord_t thickness, const size_t slice_layer_cou
     // Use a signed type for the loop counter so MSVC compiles (because it uses OpenMP 2.0, an old version).
     for (int layer_nr = 0; layer_nr < static_cast<int>(layers_ref.size()); layer_nr++)
     {
-        const coord_t xy_offset = mesh->settings.get<coord_t>((layer_nr <= layer_apply_initial_xy_offset) ? "xy_offset_layer_0" : "xy_offset");
+        const coord_t xy_offset = mesh.settings.get<coord_t>((layer_nr <= layer_apply_initial_xy_offset) ? "xy_offset_layer_0" : "xy_offset");
 
         if (xy_offset != 0)
         {
             layers_ref[layer_nr].polygons = layers_ref[layer_nr].polygons.offset(xy_offset);
         }
     }
 
-    mesh->expandXY(mesh->settings.get<coord_t>("xy_offset"));
-    log("slice make polygons took %.3f seconds\n", slice_timer.restart());
+    mesh.expandXY(mesh.settings.get<coord_t>("xy_offset"));
 }
 
-coord_t Slicer::interpolate(const coord_t x, const coord_t x0, const coord_t x1, const coord_t y0, const coord_t y1) const
+
+std::vector<std::pair<int32_t, int32_t>> Slicer::buildZHeightsForFaces(const Mesh &mesh) 
 {
-    const coord_t dx_01 = x1 - x0;
-    coord_t num = (y1 - y0) * (x - x0);
-    num += num > 0 ? dx_01 / 2 : -dx_01 / 2; // add in offset to round result
-    return y0 + num / dx_01;
+    std::vector<std::pair<int32_t, int32_t>> zHeights;
+    zHeights.reserve(mesh.faces.size());
+    for(const auto& face : mesh.faces)
+    {
+        //const MeshFace& face = mesh.faces[mesh_idx];
+        const MeshVertex& v0 = mesh.vertices[face.vertex_index[0]];
+        const MeshVertex& v1 = mesh.vertices[face.vertex_index[1]];
+        const MeshVertex& v2 = mesh.vertices[face.vertex_index[2]];
+
+        // get all vertices represented as 3D point
+        Point3 p0 = v0.p;
+        Point3 p1 = v1.p;
+        Point3 p2 = v2.p;
+
+        // find the minimum and maximum z point		
+        int32_t minZ = p0.z;
+        if (p1.z < minZ) minZ = p1.z;
+        if (p2.z < minZ) minZ = p2.z;
+
+        int32_t maxZ = p0.z;
+        if (p1.z > maxZ) maxZ = p1.z;
+        if (p2.z > maxZ) maxZ = p2.z;
```

Comment:
Code style: Brackets on new line.

Reply:
Do you a style guide where I can look into?

Reply:
Fixed

Reply:
The code style guidelines are here: https://github.com/Ultimaker/Meta